apps/dashr-financial-report/app.R

```diff
@@ -0,0 +1,1093 @@
+appName <- Sys.getenv("DASH_APP_NAME")
+
+if (appName != ""){
+
+  pathPrefix <- sprintf("/%s/", appName)
+
+
+
+  Sys.setenv(DASH_ROUTES_PATHNAME_PREFIX = pathPrefix,
+
+             DASH_REQUESTS_PATHNAME_PREFIX = pathPrefix)
+
+
+
+  setwd(sprintf("/app/apps/%s", appName))
+
+}
+
+
+
+#Source assets
+source("assets/FinancialFunctions.R")
+
+# Load Necessary Packages
+library('dash')
+library('dashCoreComponents')
+library('dashHtmlComponents')
+library('plyr')
+library('plotly')
+library('lubridate')
+
+#################################################################################
+
+app <- Dash$new()
+
+
+grey_line <- htmlDiv(list(
+  htmlHr(className = "greyline")
+))
+
+
+overview <- htmlDiv(list(
+  Header()
+))
+
+
+
+#################################################################################
+
+colors <- list(
+  background = '#ffffff',
+  text = '#111111'
+)
+
+
+
+#################################################################################
+
+#Intro Paragraph and Overview
+
+paragraph <- htmlDiv(list(
+  htmlDiv(list(
+    htmlH4('Product Summary', style = list("color" = "#ffffff")),
+    
+    htmlP("\
+                            As the industry's first index fund for individual investors, \
+                            the Calibre Index Fund is a low-cost way to gain diversified exposure \
+                            to the U.S. equity market. The fund offers exposure to 500 of the \
+                            largest U.S. companies, which span many different industries and \
+                            account for about three-fourths of the U.S. stock market's value. \
+                            The key risk for the fund is the volatility that comes with its full \
+                            exposure to the stock market. Because the Calibre Index Fund is broadly \
+                            diversified within the large-capitalization market, it may be \
+                            considered a core equity holding in a portfolio.")), className = 'rowrow2'
+  )), className = "product"
+)
+
+
+#################################################################################
+
+
+#Code for the Average Annual Performance Bar Charts
+
+
+
+ax <- list(
+  title = ""
+)
+
+years <- c("1 year", "3 year", "5 year", "10 year", "41 year")
+SP_Index <- c(21.83, 11.41, 15.79, 8.50, 0)
+Index_Fund <- c(21.67, 11.26, 15.62, 8.37, 11.11)
+
+
+performance_data <- data.frame(years, SP_Index, Index_Fund)
+
+performance_data$years <- factor(performance_data$years, levels = performance_data$years[order(c(1,2,3,4,5))])
+
+performance <- plot_ly(performance_data, x= ~years, y= ~Index_Fund, type = "bar",

```

Comment:
Long lines like this might be more digestible if you format them differently, maybe along the lines of:

```
performance <- plot_ly(
  performance_data, x= ~years, y= ~Index_Fund, 
  type = "bar", name = 'Calibre Index Fund', 
  width = 340,  height = 200) %>%
    add_trace(y= ~SP_Index, name = "S&P Index Fund") %>%
    layout(
      yaxis = ax, xaxis = ax, 
      colorway = c('#98151B', '#DCDCDC'), 
      legend = list(
        x = 1.0, y =  0.95, 
        orientation = 'h', 
        yanchor = "top", 
        font = list(size =9)
      ),
      autosize = FALSE, 
      bargap = 0.35, 
      hovermode = "closest",  
      margin = list(r=0, t=20, b=10, l=10)
    )
```

Reply:
Fixed, adjusting formatting of all lines exceeding that length (except for paragraph text for `dccMarkdown` formatting purposes).